pytorch_lightning/plugins/environments/slurm_environment.py

```diff
@@ -69,8 +66,19 @@ def master_port(self) -> int:
 
         return int(default_port)
 
-    def world_size(self):
-        return None
+    def world_size(self) -> int:
+        return int(os.environ["SLURM_NTASKS"])
+
+    def set_world_size(self, size: int) -> None:
```

Comment:
Maybe better raise an Error here to make that more explicit?

Because if I call a function like this and it doesn't error I expect the new rank to be set.